src/sdk/namespace_fs.js

```diff
@@ -924,6 +919,24 @@ class NamespaceFS {
         }
     }
 
+    async validate_results(results, fs_context) {
```

Comment:
if you remove from the list right before it's being returned to the user, you might be left with less than 1000 objects, even if you have access to more than 1000 objects, can you please check that?